### ISSUE TITLE
scv.pl.scatter for categorical annotations with Nans

### ISSUE THREAD

**Issue by Marius1311** [theislab/scvelo#117]
In previous versions of scvelo, when I plotted a categorical annotation using `scv.pl.scatter`, cells annotated with a Nan got a grey color. In the current scvelo version, cells which are annotated with Nan are added to the last annotated cluster. 

To give an example, this is what I get when I run `scv.pl.scatter(adata, color='rc_for', legend_loc='right margin'):
![image](https://user-images.githubusercontent.com/24494650/69669285-8d7df080-105f-11ea-919f-9098164bef8a.png)

when I include the `groups` keyword, things work again as expected:
`scv.pl.scatter(adata, color='rc_for', legend_loc='right margin', groups=[1, 2, 3, 4, 5])`
![image](https://user-images.githubusercontent.com/24494650/69669377-b9997180-105f-11ea-8568-627203612e40.png)

**Comment by VolkerBergen** [comment]
Hi @Marius1311 it works for me and I cannot reproduce your plot. Are you working with the latest version from source?

**Comment by Marius1311** [comment]
Hi @VolkerBergen , thanks for your swift reply.  Here is an example of what I mean:

```python
# load in scvelo from source, this is the latest version
import sys
sys.path.insert(0, '../scvelo')
import scvelo as scv
import scanpy as sc
import numpy as np

# import data, preprocess
adata = scv.datasets.forebrain()
scv.pp.filter_and_normalize(adata, n_top_genes=2000)
sc.tl.pca(adata)

# plot the cluster annotations
scv.pl.scatter(adata, basis='pca', color='Clusters', legend_loc='right margin')

# set one cluster to np.nan
adata.obs.loc[adata.obs['Clusters'] == 0, 'Clusters'] = np.nan

# plot again
scv.pl.scatter(adata, basis='pca', color='Clusters', legend_loc='right margin')
```

Am I making any mistakes here?

**Comment by Marius1311** [comment]
First plotting command gives
![image](https://user-images.githubusercontent.com/24494650/69740331-37648800-1107-11ea-881e-cf30854c9110.png)

second one gives
![image](https://user-images.githubusercontent.com/24494650/69740347-3cc1d280-1107-11ea-8f65-747aade8694a.png)

**Comment by stefanpeidli** [comment]
Hey Marius :)
I could reproduce this bug. The reason is that internally NaNs are mapped to the color of the last category which of course is dumb.

@VolkerBergen 
The bug lies in the function `get_colors` in pl.utils, specifically in the last line of that function.
This can be easily fixed (I opened a [PR#123](https://github.com/theislab/scvelo/pull/123)).

**Comment by Marius1311** [comment]
Thanks Stefan!
